src/arch/LowLevelWindow/LowLevelWindow_SDL.cpp

```diff
@@ -0,0 +1,258 @@
+#include "global.h"
+#include "LowLevelWindow_SDL.h"
+#include "arch/arch_default.h"
+#include "RageLog.h"
+
+#include <SDL3/SDL.h>
+
+#include "DisplaySpec.h"
+#include "Preference.h"
+#include "RageSurface.h"
+#include "RageSurface_Load.h"
+
+Preference<bool> g_bSDLLinuxUseWayland("SDLLinuxUseWayland", true);
+
+LowLevelWindow_SDL::LowLevelWindow_SDL()
+{
+    LOG->Info("[LLW_SDL] Using LowLevelWindow_SDL");
+    m_pWindow = nullptr;
+    m_pGlContext = nullptr;
+
+#if defined(LINUX)
+    // Nudge SDL3 to use wayland if requested (this is only needed until SDL uses wayland by default)
+    if (g_bSDLLinuxUseWayland.Get())
+    {
+        SDL_SetHint(SDL_HINT_VIDEO_DRIVER, "wayland");
+    }
+#endif
```

Comment:
By default this would use x11 unless they specify an environment hint (baked into SDL) or flip this option to true. When they use wayland by default in the (hopefully) near future, we shouldn't have to do anything special.